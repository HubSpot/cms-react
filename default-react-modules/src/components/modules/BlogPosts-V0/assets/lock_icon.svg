<svg width="31" height="32" viewBox="0 0 31 32" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M26.4484 11.08H25.2198V9.8515C25.2198 4.41116 20.8087 0 15.3683 0C9.928 0 5.51684 4.41116 5.51684 9.8515V11.08H4.31148C1.93089 11.08 0 13.0098 0 15.3915V27.6885C0 30.0703 1.93089 32 4.31148 32H26.4484C28.8301 32 30.7599 30.0691 30.7599 27.6885V15.3915C30.7599 13.0098 28.829 11.08 26.4484 11.08V11.08ZM19.0656 19.7146C19.0656 21.0718 17.9657 22.1717 16.6085 22.1717V25.8573C16.6085 26.5353 16.058 27.0858 15.3799 27.0858C14.7019 27.0858 14.1514 26.5353 14.1514 25.8573V22.1601C12.7907 22.1601 11.6885 21.0579 11.6885 19.6972C11.6885 18.3365 12.7907 17.2343 14.1514 17.2343H16.6085C17.9633 17.2598 19.0528 18.3586 19.0656 19.7134V19.7146ZM9.22564 9.8515C9.22564 6.45215 11.9806 3.69721 15.3799 3.69721C18.7793 3.69721 21.5342 6.45215 21.5342 9.8515V11.08H9.22564V9.8515Z" fill="#425B76"/>
</svg>
