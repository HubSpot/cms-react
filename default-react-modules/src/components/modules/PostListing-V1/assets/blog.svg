<?xml version="1.0" encoding="UTF-8"?>
<svg width="32px" height="28px" viewBox="0 0 32 28" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.1 (43504) - http://www.bohemiancoding.com/sketch -->
    <title>Untitled 2</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="high-blog" fill-rule="nonzero" fill="#3F5D75">
            <path d="M28.22,0.81 L8.67,0.81 C6.781,0.81 5.25,2.341 5.25,4.23 L5.25,23.29 C5.25,23.842 4.802,24.29 4.25,24.29 C3.698,24.29 3.25,23.842 3.25,23.29 L3.25,23.29 L3.25,9.12 C3.185,8.362 2.554,7.771 1.785,7.771 C1.016,7.771 0.385,8.362 0.32,9.114 L0.32,9.119 L0.32,23.289 C0.328,25.36 1.948,27.05 3.989,27.169 L4,27.169 C4.1,27.169 4.19,27.169 4.3,27.169 L26.3,27.169 C29.271,27.169 31.68,24.76 31.68,21.789 L31.68,21.789 L31.68,4.229 C31.68,4.229 31.68,4.229 31.68,4.229 C31.68,2.34 30.149,0.809 28.26,0.809 C28.246,0.809 28.232,0.809 28.218,0.809 L28.22,0.809 L28.22,0.81 Z M28.71,21.81 C28.704,23.155 27.615,24.244 26.271,24.25 L8.18,24.25 L8.18,4.25 C8.18,3.979 8.399,3.76 8.67,3.76 L8.67,3.76 L28.22,3.76 C28.491,3.76 28.71,3.979 28.71,4.25 L28.71,4.25 L28.71,21.81 Z" id="Shape"></path>
            <path d="M11.61,5.7 L25.29,5.7 C26.102,5.7 26.76,6.358 26.76,7.17 L26.76,11.07 C26.76,11.882 26.102,12.54 25.29,12.54 L11.61,12.54 C10.798,12.54 10.14,11.882 10.14,11.07 L10.14,7.17 C10.14,6.358 10.798,5.7 11.61,5.7 Z" id="Shape"></path>
            <path d="M25.28,14.49 L11.6,14.49 C10.842,14.555 10.251,15.186 10.251,15.955 C10.251,16.724 10.842,17.355 11.594,17.42 L11.599,17.42 L25.279,17.42 C25.315,17.423 25.357,17.425 25.4,17.425 C26.212,17.425 26.87,16.767 26.87,15.955 C26.87,15.143 26.212,14.485 25.4,14.485 C25.357,14.485 25.315,14.487 25.273,14.49 L25.278,14.49 L25.28,14.49 Z" id="Shape"></path>
            <path d="M25.28,19.38 L11.6,19.38 C11.564,19.377 11.521,19.375 11.479,19.375 C10.667,19.375 10.009,20.033 10.009,20.845 C10.009,21.657 10.667,22.315 11.479,22.315 C11.522,22.315 11.564,22.313 11.606,22.31 L11.601,22.31 L25.281,22.31 C25.317,22.313 25.359,22.315 25.402,22.315 C26.214,22.315 26.872,21.657 26.872,20.845 C26.872,20.033 26.214,19.375 25.402,19.375 C25.359,19.375 25.317,19.377 25.275,19.38 L25.28,19.38 L25.28,19.38 Z" id="Shape"></path>
        </g>
    </g>
</svg>
