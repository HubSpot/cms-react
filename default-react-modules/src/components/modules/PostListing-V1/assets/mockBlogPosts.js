export const mockBlogPosts = [
  {
    id: 1,
    name: 'How to break your ankle and still smile',
    absoluteUrl: 'https://test-domain/blog/01-01-2023/broken-ankles',
    featuredImage:
      'https://www.mayoclinic.org/-/media/kcms/gbs/patient-consumer/images/2013/08/26/11/10/ds00951_im03497_r7_footanklethu_jpg.jpg',
    featuredImageAltText: 'Bone structure of a foot',
    blogAuthor: {
      avatar:
        'https://cdn3.iconfinder.com/data/icons/avatars-9/145/Avatar_Cat-512.png',
      name: 'Brandon Van Dyck',
      displayName: 'Brandon Van Dyck',
      slug: 'brandonvandyck',
    },
    tagList: [
      {
        name: 'Feet',
        slug: 'feet',
      },
    ],
    parentBlog: {},
    publishDate: 1756388800000,
    publishDateLocalized: { date: 'Jun 30, 2022 12:00:00 AM' },
    postListContent:
      "<h3>Ankles may break, but spirits don't have to</h3> <p>Find out why a broken bone does not mean frowns all around.</p>",
  },
  {
    id: 2,
    name: 'E-Bikes: A Guide to Building and Riding',
    absoluteUrl: 'https://test-domain/blog/01-02-2023/ebikes',
    featuredImage:
      'https://pedegoelectricbikes.com/wp-content/uploads/2020/12/pedego-ridge-rider-classic-600x494.jpg',
    featuredImageAltText: 'Electric bike',
    blogAuthor: {
      avatar:
        'https://cdn3.iconfinder.com/data/icons/avatars-9/145/Avatar_Cat-512.png',
      name: 'Joey Blake',
      displayName: 'Joey Blake',
      slug: 'jblake',
    },
    tagList: [
      {
        name: 'Bikes',
        slug: 'bikes',
      },
      {
        name: 'Electricity',
        slug: 'electricity',
      },
    ],
    parentBlog: {},
    publishDate: 1756388800000,
    publishDateLocalized: { date: 'Jun 30, 2022 12:00:00 AM' },
    postListContent:
      '<h3>Bikes were traditionally powered by calories -- no more!</h3> <p>Experience the joy of electricity powered movement.</p>',
  },
  {
    id: 3,
    name: 'Mesh Wifi: Boost your home wifi experience',
    absoluteUrl: 'https://test-domain/blog/01-03-2023/wifi',
    featuredImage:
      'https://m.media-amazon.com/images/I/41bMo8AYiML._AC_SY1000_.jpg',
    featuredImageAltText: 'Home with mesh wifi ',
    blog_avatar: 'https://cdn-icons-png.flaticon.com/512/3093/3093444.png',
    blogAuthor: {
      avatar:
        'https://cdn3.iconfinder.com/data/icons/avatars-9/145/Avatar_Cat-512.png',
      name: 'Wifi Willy',
      displayName: 'Wilfred Wilhelm Wifi',
      slug: 'wwifi',
    },
    tagList: [
      {
        name: 'Wifi',
        slug: 'wifi',
      },
      {
        name: 'Mesh',
        slug: 'mesh',
      },
    ],
    parentBlog: {},
    publishDate: 1756388800000,
    publishDateLocalized: { date: 'Jun 30, 2022 12:00:00 AM' },
    postListContent:
      '<h3>【Eliminate WiFi Dead Spots】Cover more than 6000 square feet from garage to backyard, seamless single WiFi name for whole house.</h3> <p>Easy to expand the coverage by install 3 pack mesh system, simply setup mesh to eliminate WiFi dead spots. Support up to 9 dots to build WiFi system for any home size</p>',
  },
  {
    id: 4,
    name: 'Being a Chef: To live amongst flame',
    absoluteUrl: 'https://test-domain/blog/01-04-2023/being-a-chef',
    featuredImage:
      'https://www.finedininglovers.com/sites/g/files/xknfdk626/files/2021-07/chef%20%281%29.jpg',
    featuredImageAltText: 'Chef using alcohol and fire to cook a dish',
    blog_avatar: 'https://cdn-icons-png.flaticon.com/512/3093/3093444.png',
    blogAuthor: {
      avatar: '',
      name: 'Chef Knife',
      displayName: 'Chefred Knifer',
      slug: 'chefredknifer',
    },
    tagList: [
      {
        name: 'Chef',
        slug: 'chef',
      },
      {
        name: 'Fire',
        slug: 'fire',
      },
    ],
    parentBlog: {},
    publishDate: 1756388800000,
    publishDateLocalized: { date: 'Jun 30, 2022 12:00:00 AM' },
    postListContent:
      "<h3>Being a Chef</h3> <p>As much as restaurants are driven by chefs' desire to feed and entertain, and to express themselves, to produce the best possible food, delivered with impeccable service, they are, when it comes down to it, businesses and every chef that wants to be successful must get a handle on that side of things. It is perhaps the most important factor for longevity.</p>",
  },
];
