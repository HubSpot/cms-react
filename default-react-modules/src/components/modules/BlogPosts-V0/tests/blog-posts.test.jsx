import { afterEach, describe, expect, it } from 'vitest';
import { cleanup, render, screen } from '@testing-library/react';
import { mockBlogPosts } from '../assets/mockData';

import { Component as BlogPosts } from '../index.jsx';

const defaultProps = {
  hublParameters: {
    blogPosts: mockBlogPosts,
  },
  layout: 'grid',
  columns: 3,
  alternateImage: false,
  fullImage: false,
  displayForEachListItem: ['title', 'image', 'authorName', 'authorImage'],
  groupStyle: {
    groupBackgroundImage: {
      spacing: {},
    },
    groupImage: {
      groupSize: {
        width: '25',
      },
      groupSpacing: {
        betweenImageAndContent: '20',
        spacing: {},
      },
      groupCorner: {
        radius: 0,
      },
    },
    groupPost: {
      groupSpacing: {
        spacing: {},
      },
      groupBackground: {},
      groupBorder: {
        border: {},
      },
      groupCorner: {
        radius: 0,
      },
    },
    groupTitle: {
      headingLevel: 'h2',
      groupText: {
        font: {
          color: '',
          size: '16',
          size_unit: 'px',
        },
      },
      groupHover: {
        groupText: {
          font: {
            color: '',
            size: '16',
            size_unit: 'px',
          },
        },
      },
      groupSpacing: {
        spacing: {},
      },
    },
    groupAuthor: {
      groupImage: {
        groupCorner: {
          radius: 0,
        },
      },
      groupText: {
        font: {
          color: '',
          size: '16',
          size_unit: 'px',
        },
      },
      groupHover: {
        groupText: {
          font: {
            color: '',
            size: '16',
            size_unit: 'px',
          },
        },
      },
      groupSpacing: {
        spacing: {},
      },
    },
    groupTags: {
      groupText: {
        font: {},
      },
      groupBorder: {
        border: {},
      },
      groupSpacing: {
        spacing: {},
      },
      groupBackground: {},
      groupCorner: {
        radius: 0,
      },
      groupHover: {
        groupText: {
          font: {},
        },
        groupBorder: {
          border: {},
        },
        groupBackground: {
          color: '',
        },
      },
    },
    groupDescription: {
      groupText: {
        font: {},
      },
      groupSpacing: {
        spacing: {},
      },
    },
    groupButton: {
      groupText: {
        font: {},
      },
      groupBorder: {
        border: {},
      },
      groupCorner: {
        radius: 0,
      },
      groupSpacing: {
        spacing: {},
      },
      groupBackground: {},
      groupHover: {
        groupText: {
          font: {},
        },
        groupBorder: {
          border: {},
        },
        groupBackground: {},
      },
    },
    groupPublishDate: {
      groupText: {
        font: {},
      },
      groupSpacing: {
        spacing: {},
      },
    },
    groupContent: {
      spacing: {},
    },
  },
  groupDefaultText: {
    featuredImageText: 'Featured image text',
  },
};

describe('BlogPosts', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders BlogPosts', () => {
    const { asFragment } = render(<BlogPosts {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('when fullImage is set to true', () => {
    it('puts the image as background image', () => {
      render(<BlogPosts {...defaultProps} fullImage={true} />);

      const blogPostsWithBackgroundImage = screen.getAllByTestId(
        'blog-post-with-background-image'
      );
      expect(blogPostsWithBackgroundImage.length).toBe(4);
    });
  });

  describe('when showAuthorImage is false', () => {
    it('does not show the author avatar', () => {
      render(
        <BlogPosts
          {...defaultProps}
          displayForEachListItem={['image', 'title', 'authorName']}
        />
      );

      const blogPostAuthorAvatar = screen.queryAllByTestId('author-avatar');
      expect(blogPostAuthorAvatar.length).toBe(0);
    });
  });

  describe('when showAuthorName is false', () => {
    it('does not show the author name', () => {
      render(
        <BlogPosts
          {...defaultProps}
          displayForEachListItem={['image', 'title', 'authorImage']}
        />
      );

      const blogPostAuthorName = screen.queryAllByTestId('author-name');
      expect(blogPostAuthorName.length).toBe(0);
    });
  });

  describe('when showAuthorImage and showAuthorName are false', () => {
    it('does not render the author section', () => {
      render(
        <BlogPosts
          {...defaultProps}
          displayForEachListItem={['image', 'title']}
        />
      );

      const blogPostAuthorWrapper = screen.queryAllByTestId('author-wrapper');
      expect(blogPostAuthorWrapper.length).toBe(0);
    });
  });

  describe('Grid layout', () => {
    it('properly calculates grid styling when space between posts is present', () => {
      const blogPostWithSpaceBetweenPosts = {
        ...defaultProps,
        groupStyle: {
          ...defaultProps.groupStyle,
          groupPost: {
            groupBorder: {
              border: {},
            },
            groupCorner: {
              radius: 0,
            },
            groupBackground: {},
            groupSpacing: {
              spaceBetweenPosts: '30',
              spacing: {},
            },
          },
        },
      };

      render(<BlogPosts {...blogPostWithSpaceBetweenPosts} />);
      const blogPosts = screen.queryAllByTestId('blog-post');

      const blogPostStyle = blogPosts[0].style;

      expect(blogPostStyle.getPropertyValue('--grid-post-width')).toEqual(
        'calc(50% - 15px)'
      );
      expect(
        blogPostStyle.getPropertyValue('--grid-space-between-posts')
      ).toEqual('30px');
      expect(
        blogPostStyle.getPropertyValue('--grid-post-width-three-column')
      ).toEqual('calc(33.3% - 20px)');
      expect(
        blogPostStyle.getPropertyValue('--grid-post-width-four-column')
      ).toEqual('calc(25% - 22.5px)');
    });

    it('uses a static calcualtion for grid styling when space between posts is not present', () => {
      render(<BlogPosts {...defaultProps} />);
      const blogPosts = screen.queryAllByTestId('blog-post');
      const blogPostStyle = blogPosts[0].style;

      expect(blogPostStyle.getPropertyValue('--grid-post-width')).toEqual(
        'calc(50% - 0.5rem)'
      );
      expect(
        blogPostStyle.getPropertyValue('--grid-space-between-posts')
      ).toEqual('1rem');
      expect(
        blogPostStyle.getPropertyValue('--grid-post-width-three-column')
      ).toEqual('calc(33.3% - 0.66rem)');
      expect(
        blogPostStyle.getPropertyValue('--grid-post-width-four-column')
      ).toEqual('calc(25% - 0.75rem)');
    });
  });

  describe('Side by side layout', () => {
    const blogPostsWithSideBySideLayout = {
      ...defaultProps,
      layout: 'sideBySide',
    };

    it('properly calculates width for image and content when imageWidth and betweenImageAndContent present', () => {
      render(<BlogPosts {...blogPostsWithSideBySideLayout} />);
      const blogPosts = screen.queryAllByTestId('blog-post');
      const blogPostStyle = blogPosts[0].style;

      expect(blogPostStyle.getPropertyValue('--image-wrapper-width')).toEqual(
        'calc(25% - 10px)'
      );
      expect(blogPostStyle.getPropertyValue('--content-wrapper-width')).toEqual(
        'calc(75% - 10px)'
      );
    });

    it('properly calculates width for image and content when only imageWidth is present', () => {
      const blogPostsWithNoSpaceBetweenContent = {
        ...blogPostsWithSideBySideLayout,
        groupStyle: {
          ...defaultProps.groupStyle,
          groupImage: {
            groupSize: {
              width: '25',
            },
            groupCorner: {
              radius: 0,
            },
            groupSpacing: {
              spacing: {},
            },
          },
        },
      };

      render(<BlogPosts {...blogPostsWithNoSpaceBetweenContent} />);
      const blogPosts = screen.queryAllByTestId('blog-post');
      const blogPostStyle = blogPosts[0].style;

      expect(blogPostStyle.getPropertyValue('--image-wrapper-width')).toEqual(
        'calc(25% - 0.5rem)'
      );
      expect(blogPostStyle.getPropertyValue('--content-wrapper-width')).toEqual(
        'calc(75% - 0.5rem)'
      );
    });

    it('properly calculates width for image and content when only betweenImageAndContent is present', () => {
      const blogPostsWithNoImageWidth = {
        ...blogPostsWithSideBySideLayout,
        groupStyle: {
          ...defaultProps.groupStyle,
          groupImage: {
            groupSize: {},
            groupCorner: {
              radius: 0,
            },
            groupSpacing: {
              betweenImageAndContent: '20',
              spacing: {},
            },
          },
        },
      };

      render(<BlogPosts {...blogPostsWithNoImageWidth} />);
      const blogPosts = screen.queryAllByTestId('blog-post');
      const blogPostStyle = blogPosts[0].style;

      expect(blogPostStyle.getPropertyValue('--image-wrapper-width')).toEqual(
        'calc(40% - 10px)'
      );
      expect(blogPostStyle.getPropertyValue('--content-wrapper-width')).toEqual(
        'calc(60% - 10px)'
      );
    });

    it('uses a static calculation when neither imageWidth or betweenImageAndContent are present', () => {
      const blogPostsWithNoImageWidthOrSpaceBetweenContent = {
        ...blogPostsWithSideBySideLayout,
        groupStyle: {
          ...defaultProps.groupStyle,
          groupImage: {
            groupSize: {},
            groupCorner: {
              radius: 0,
            },
            groupSpacing: {
              spacing: {},
            },
          },
        },
      };

      render(<BlogPosts {...blogPostsWithNoImageWidthOrSpaceBetweenContent} />);
      const blogPosts = screen.queryAllByTestId('blog-post');
      const blogPostStyle = blogPosts[0].style;

      expect(blogPostStyle.getPropertyValue('--image-wrapper-width')).toEqual(
        'calc(40% - 0.5rem)'
      );
      expect(blogPostStyle.getPropertyValue('--content-wrapper-width')).toEqual(
        'calc(60% - 0.5rem)'
      );
    });

    it('properly sets the order and margin when imageSide is left', () => {
      render(<BlogPosts {...blogPostsWithSideBySideLayout} />);
      const blogPosts = screen.queryAllByTestId('blog-post-image');
      const leftSideBlogPostStyle = blogPosts[0].style;

      expect(leftSideBlogPostStyle.getPropertyValue('--margin-right')).toEqual(
        '20px'
      );
      expect(leftSideBlogPostStyle.getPropertyValue('--order')).toEqual('1');
    });

    it('properly sets the order and margin when imageSide is right', () => {
      const blogPostsWithAlternatingImage = {
        ...blogPostsWithSideBySideLayout,
        alternateImage: true,
      };

      render(<BlogPosts {...blogPostsWithAlternatingImage} />);
      const blogPosts = screen.queryAllByTestId('blog-post-image');
      const rightSideBlogPostStyle = blogPosts[1].style;

      expect(rightSideBlogPostStyle.getPropertyValue('--margin-left')).toEqual(
        '20px'
      );
      expect(rightSideBlogPostStyle.getPropertyValue('--order')).toEqual('2');
    });
  });
});
