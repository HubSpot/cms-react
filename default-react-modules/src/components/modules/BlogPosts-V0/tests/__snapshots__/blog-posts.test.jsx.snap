// Vitest Snapshot v1

exports[`BlogPosts > renders BlogPosts 1`] = `
<DocumentFragment>
  <style>
    
            ._hsBlogPost_ab621b {
      border-radius: 0px;
    }
    ._hsBlogPostImageWrapper_ab621b {
    }
    ._hsBlogPostImageOverlay_ab621b {
    }
    ._hsBlogPostListingContent_ab621b {
    }
    ._hsBlogPostTitle_ab621b {
    }
    ._hsBlogPostTitleLink_ab621b {
    }
    ._hsBlogPostTitleLink_ab621b:hover {
    }
    ._hsBlogPostAuthorImageWrapper_ab621b {
      border-radius: 0px;
    }
    ._hsBlogPostAuthorImage_ab621b {
      width: undefinedpx;
    }
    ._hsBlogPostAuthor_ab621b {
    }
    ._hsBlogPostAuthor_ab621b:hover {
    }
    ._hsBlogPostTag_ab621b {
      border-radius: 0px;
    }
    ._hsBlogPostTag_ab621b:hover {
    }
    ._hsBlogPostTag_ab621b:not(:last-child) {
      margin-right ;
    }
    ._hsBlogPostPublishDate_ab621b {
    }
    ._hsBlogPostDescription_ab621b {
    }
    ._hsBlogPostButtonWrapper_ab621b {
    }
    ._hsBlogPostButton_ab621b {
      border-radius: 0px;
    }
    ._hsBlogPostButton_ab621b:hover {
    }
  
      
  </style>
  <section
    class="_hsBlogPostListing_ab621b _hsBlogPostGrid_ab621b _hsBlogPostGridColumn3_ab621b"
  >
    <article
      class="_hsBlogPost_ab621b _hsBlogPostListingPostColumn3_ab621b"
      data-testid="blog-post"
      style="--grid-post-width: calc(50% - 0.5rem); --grid-space-between-posts: 1rem; --grid-post-width-three-column: calc(33.3% - 0.66rem); --grid-post-width-four-column: calc(25% - 0.75rem);"
    >
      <a
        aria-label="Featured image text Bone structure of a foot"
        class="_hsBlogPostImageWrapper_ab621b"
        data-testid="blog-post-image"
        href="https://test-domain/blog/01-01-2023/broken-ankles"
        style="--margin-right: 20px; --order: 1; --border-radius: 0px;"
      >
        <img
          alt="Bone structure of a foot"
          class="_hsBlogPostListingImage_ab621b"
          height="585"
          loading="eager"
          src="https://www.mayoclinic.org/-/media/kcms/gbs/patient-consumer/images/2013/08/26/11/10/ds00951_im03497_r7_footanklethu_jpg.jpg"
          width="632"
        />
      </a>
      <div
        class="_hsBlogPostListingContent_ab621b"
      >
        <h2
          class="_hsBlogPostTitle_ab621b"
        >
          <a
            class="_hsBlogPostTitleLink_ab621b"
            href="https://test-domain/blog/01-01-2023/broken-ankles"
          >
            How to break your ankle and still smile
          </a>
        </h2>
        <div
          class="_hsBlogPostAuthor_ab621b"
          data-testid="author-wrapper"
        >
          <div
            class="_hsBlogPostAuthorImageWrapper_ab621b"
          >
            <img
              class="_hsBlogPostAuthorImage_ab621b"
              data-testid="author-avatar"
              loading="lazy"
              src="https://cdn3.iconfinder.com/data/icons/avatars-9/145/Avatar_Cat-512.png"
            />
          </div>
          <a
            class="_hsBlogPostAuthor_ab621b"
            data-testid="author-name"
            href="undefined/author/brandonvandyck"
          >
            Brandon Van Dyck
          </a>
        </div>
      </div>
    </article>
    <article
      class="_hsBlogPost_ab621b _hsBlogPostListingPostColumn3_ab621b"
      data-testid="blog-post"
      style="--grid-post-width: calc(50% - 0.5rem); --grid-space-between-posts: 1rem; --grid-post-width-three-column: calc(33.3% - 0.66rem); --grid-post-width-four-column: calc(25% - 0.75rem);"
    >
      <a
        aria-label="Featured image text Electric bike"
        class="_hsBlogPostImageWrapper_ab621b"
        data-testid="blog-post-image"
        href="https://test-domain/blog/01-02-2023/ebikes"
        style="--margin-right: 20px; --order: 1; --border-radius: 0px;"
      >
        <img
          alt="Electric bike"
          class="_hsBlogPostListingImage_ab621b"
          height="494"
          loading="eager"
          src="https://pedegoelectricbikes.com/wp-content/uploads/2020/12/pedego-ridge-rider-classic-600x494.jpg"
          width="600"
        />
      </a>
      <div
        class="_hsBlogPostListingContent_ab621b"
      >
        <h2
          class="_hsBlogPostTitle_ab621b"
        >
          <a
            class="_hsBlogPostTitleLink_ab621b"
            href="https://test-domain/blog/01-02-2023/ebikes"
          >
            E-Bikes: A Guide to Building and Riding
          </a>
        </h2>
        <div
          class="_hsBlogPostAuthor_ab621b"
          data-testid="author-wrapper"
        >
          <div
            class="_hsBlogPostAuthorImageWrapper_ab621b"
          >
            <img
              class="_hsBlogPostAuthorImage_ab621b"
              data-testid="author-avatar"
              loading="lazy"
              src="https://cdn3.iconfinder.com/data/icons/avatars-9/145/Avatar_Cat-512.png"
            />
          </div>
          <a
            class="_hsBlogPostAuthor_ab621b"
            data-testid="author-name"
            href="undefined/author/jblake"
          >
            Joey Blake
          </a>
        </div>
      </div>
    </article>
    <article
      class="_hsBlogPost_ab621b _hsBlogPostListingPostColumn3_ab621b"
      data-testid="blog-post"
      style="--grid-post-width: calc(50% - 0.5rem); --grid-space-between-posts: 1rem; --grid-post-width-three-column: calc(33.3% - 0.66rem); --grid-post-width-four-column: calc(25% - 0.75rem);"
    >
      <a
        aria-label="Featured image text Home with mesh wifi "
        class="_hsBlogPostImageWrapper_ab621b"
        data-testid="blog-post-image"
        href="https://test-domain/blog/01-03-2023/wifi"
        style="--margin-right: 20px; --order: 1; --border-radius: 0px;"
      >
        <img
          alt="Home with mesh wifi "
          class="_hsBlogPostListingImage_ab621b"
          height="500"
          loading="eager"
          src="https://m.media-amazon.com/images/I/41bMo8AYiML._AC_SY1000_.jpg"
          width="500"
        />
      </a>
      <div
        class="_hsBlogPostListingContent_ab621b"
      >
        <h2
          class="_hsBlogPostTitle_ab621b"
        >
          <a
            class="_hsBlogPostTitleLink_ab621b"
            href="https://test-domain/blog/01-03-2023/wifi"
          >
            Mesh Wifi: Boost your home wifi experience
          </a>
        </h2>
        <div
          class="_hsBlogPostAuthor_ab621b"
          data-testid="author-wrapper"
        >
          <div
            class="_hsBlogPostAuthorImageWrapper_ab621b"
          >
            <img
              class="_hsBlogPostAuthorImage_ab621b"
              data-testid="author-avatar"
              loading="lazy"
              src="https://cdn3.iconfinder.com/data/icons/avatars-9/145/Avatar_Cat-512.png"
            />
          </div>
          <a
            class="_hsBlogPostAuthor_ab621b"
            data-testid="author-name"
            href="undefined/author/wwifi"
          >
            Wilfred Wilhelm Wifi
          </a>
        </div>
      </div>
    </article>
    <article
      class="_hsBlogPost_ab621b _hsBlogPostListingPostColumn3_ab621b"
      data-testid="blog-post"
      style="--grid-post-width: calc(50% - 0.5rem); --grid-space-between-posts: 1rem; --grid-post-width-three-column: calc(33.3% - 0.66rem); --grid-post-width-four-column: calc(25% - 0.75rem);"
    >
      <a
        aria-label="Featured image text Chef using alcohol and fire to cook a dish"
        class="_hsBlogPostImageWrapper_ab621b"
        data-testid="blog-post-image"
        href="https://test-domain/blog/01-04-2023/being-a-chef"
        style="--margin-right: 20px; --order: 1; --border-radius: 0px;"
      >
        <img
          alt="Chef using alcohol and fire to cook a dish"
          class="_hsBlogPostListingImage_ab621b"
          height="814"
          loading="lazy"
          src="https://www.finedininglovers.com/sites/g/files/xknfdk626/files/2021-07/chef%20%281%29.jpg"
          width="1221"
        />
      </a>
      <div
        class="_hsBlogPostListingContent_ab621b"
      >
        <h2
          class="_hsBlogPostTitle_ab621b"
        >
          <a
            class="_hsBlogPostTitleLink_ab621b"
            href="https://test-domain/blog/01-04-2023/being-a-chef"
          >
            Being a Chef: To live amongst flame
          </a>
        </h2>
        <div
          class="_hsBlogPostAuthor_ab621b"
          data-testid="author-wrapper"
        >
          <a
            class="_hsBlogPostAuthor_ab621b"
            data-testid="author-name"
            href="undefined/author/chefredknifer"
          >
            Chefred Knifer
          </a>
        </div>
      </div>
    </article>
  </section>
</DocumentFragment>
`;
